systemd/systemd.go

```diff
@@ -614,14 +645,19 @@ func (s *systemd) getUnitStatus(properties []string, unitNames []string) ([]*Uni
 				}
 			}
 			if len(missing) > 0 {
-				return nil, fmt.Errorf("cannot get unit %q status: missing %s in ‘systemctl show’ output", cur.UnitName, strings.Join(missing, ", "))
+				return nil, fmt.Errorf("cannot get unit %q status: missing %s in ‘systemctl show’ output", cur.Name, strings.Join(missing, ", "))
 			}
-			sts = append(sts, cur)
-			if len(sts) > len(unitNames) {
-				break // wut
-			}
-			if cur.UnitName != unitNames[len(sts)-1] {
-				return nil, fmt.Errorf("cannot get unit status: queried status of %q but got status of %q", unitNames[len(sts)-1], cur.UnitName)
+
+			// The 'Names' DBUS property of the 'Unit' interface exposes all
```

Comment:
Super nitpick (and sorry for not noticing it before), but since you are not making a D-Bus call, this first sentence can be a bit confusing. I'd just write "The 'Names' property exposes..."